IOSDeviceLib/Printing.cpp

```diff
@@ -1,16 +1,24 @@
+#include <mutex>
+
 #include "SocketHelper.h"
 #include "Printing.h"
 #include "Constants.h"
 #include "json.hpp"
 
+std::mutex print_mutex;
+
 void print(const char* str)
 {
+	// We need to lock the print method because in some cases we print different parts of messages
+	// from different threads.
+	print_mutex.lock();
```

Comment:
If we decide to lock printing after all can we place the mutex on the device log operations only so as to not slow down all the other operations 🤔